apps/braze/src/components/create/FieldsStep.tsx

```diff
@@ -15,37 +17,68 @@ const FieldsStep = ({
   setSelectedFields,
   handleNextStep,
 }: FieldsStepProps) => {
-  const handleFieldChange = (fieldId: string) => {
+  const allFields = entry.fields;
+  const allFieldIds = allFields.map((field) => field.uniqueId());
+  const allSelected = selectedFields.size === allFieldIds.length;
+
+  const idToLabel = Object.fromEntries(
+    allFields.map((f) => [f.uniqueId(), f.displayNameForCreate()])
+  );
+  const labelToId = Object.fromEntries(
+    allFields.map((f) => [f.displayNameForCreate(), f.uniqueId()])
+  );
+
+  const currentSelection = Array.from(selectedFields)
+    .map((id) => idToLabel[id])
+    .filter(Boolean);
+
+  const handleSelectField = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { checked, value } = event.target;
+    const id = labelToId[value];
     const newSelectedFields = new Set(selectedFields);
-    if (newSelectedFields.has(fieldId)) {
-      newSelectedFields.delete(fieldId);
+    if (checked) {
+      newSelectedFields.add(id);
     } else {
-      newSelectedFields.add(fieldId);
+      newSelectedFields.delete(id);
     }
     setSelectedFields(newSelectedFields);
   };
 
+  const handleSelectAll = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.checked) {
+      setSelectedFields(new Set(allFieldIds));
+    } else {
+      setSelectedFields(new Set());
+    }
+  };
+
   return (
     <>
       <Paragraph>
         Select the fields you would like to generate into Content Blocks. Referenced fields are not
         available in this list, but can be linked directly through the entry.
       </Paragraph>
-      <FormControl marginBottom="spacingXs">
+      <FormControl isRequired isInvalid={setSelectedFields.length === 0} marginBottom="spacing4Xl">
         <FormControl.Label>Select Fields</FormControl.Label>
-        <Select
-          value={Array.from(selectedFields)[selectedFields.size - 1] || ''}
-          onChange={(e) => handleFieldChange(e.target.value)}
-          data-testid={`select-${entry.id}`}>
-          <Option value="">Select a field</Option>
-          {entry.fields.map((field) => (
-            <Option key={field.uniqueId()} value={field.uniqueId()}>
-              {field.displayName()}
-            </Option>
+        <Multiselect
+          currentSelection={currentSelection}
+          popoverProps={{ isFullWidth: true, listMaxHeight: 108 }}
+          placeholder="Select one or more">
+          <Multiselect.SelectAll onSelectItem={handleSelectAll} isChecked={allSelected} />
```

Comment:
I would inline the allSelected boolean since it's not used in other places
```suggestion
          <Multiselect.SelectAll onSelectItem={handleSelectAll} isChecked={selectedFields.size === allFieldIds.length} />
```

Reply:
Done! ✅